tony-core/src/main/java/com/linkedin/tony/runtime/MLGenericRuntime.java

```diff
@@ -92,9 +114,29 @@ public boolean validateAndUpdateConfig(Configuration tonyConf) {
         if (!validate(tonyConf)) {
             return false;
         }
+
+        checkAndPrepareTBResource(tonyConf);
         return true;
     }
 
+    /**
+     * Set the required resources when enable auto-starting tensorboard.
+     */
+    private void checkAndPrepareTBResource(Configuration tonyConf) {
+        if (!enableSidecarTB(tonyConf)) {
+            return;
+        }
+
+        // TODO: 2021/5/8 check illegal keys. [tensorboard] role is illegal
+
+        tonyConf.set(TENSORBOARD_RESOURCE_VCORE, "1");
+        tonyConf.set(TENSORBOARD_RESOURCE_INSTANCES, "1");
+        tonyConf.set(TENSORBOARD_RESOURCE_MEMORY, "2g");
```

Comment:
create default values in configuration (like TENSORBOARD_RESOURCE_INSTANCES_DEFAULT)